LeaderboardBackend/Controllers/LeaderboardsController.cs

```diff
@@ -86,4 +86,28 @@ public async Task<ActionResult<LeaderboardViewModel>> CreateLeaderboard(
             }
         );
     }
+
+    [Authorize(Policy = UserTypes.ADMINISTRATOR)]
+    [HttpPut("leaderboard/{id:long}/restore")]
+    [SwaggerOperation("Restores a deleted leaderboard.", OperationId = "restoreLeaderboard")]
+    [SwaggerResponse(200)]
+    [SwaggerResponse(401)]
+    [SwaggerResponse(403, "The requesting `User` is unauthorized to restore `Leaderboard`s.")]
+    [SwaggerResponse(404, "The `Leaderboard` was not found, or it wasn't deleted in the first place.")]
+    public async Task<ActionResult<LeaderboardViewModel>> RestoreLeaderboard(
+        long id
+    )
+    {
+        RestoreLeaderboardResult r = await leaderboardService.RestoreLeaderboard(id);
+
+        return r.Match<ActionResult<LeaderboardViewModel>>(
+            board => Ok(LeaderboardViewModel.MapFrom(board)),
+            notFound => NotFound(),
+            neverDeleted =>
+            {
+                ModelState.AddModelError("Leaderboard", "LeaderboardWasNeverPreviouslyDeleted");
+                return NotFound(new ValidationProblemDetails(ModelState));
```

Comment:
I'm not sure that this counts as a validation error. Maybe just make a new DTO for this?

Reply:
Frankly I only did it cuz I was copying existing tests. Currently made it so that it just returns a string saying "Was never deleted". Feel free to suggest something better? I can't imagine what else to say here.

Reply:
Check [this page](https://learn.microsoft.com/en-us/aspnet/core/web-api/handle-errors) for ways to handle client errors. I recommend returning a ProblemDetails with a custom Title of NotDeleted or something similar.